web/state/Roles.js

```diff
@@ -0,0 +1,154 @@
+/* eslint-disable no-console */
+const ZERO = 0;
+const ERR_NO_ROLE_NAME = 1;
+const ERR_SPACE_IN_ROLE_LABEL = 2;
+const ERR_ROLE_ALREADY_EXISTS = 3;
+
+class Roles {
```

Comment:
Would it be better naming this RolesManager (and UsersManager)?

My first impression was that why we pass bdk as param in a Roles object. But it makes more sense after understanding this is actually served as Roles manager.